Apps/WebClient/src/ClientApp/src/router.ts

```diff
@@ -241,8 +253,17 @@ router.beforeEach(async (to, from, next) => {
         // Make sure that the route accepts the current state
         const currentUserState = calculateUserState();
         logger.debug(`current state: ${currentUserState}`);
+        const config = store.getters["config/webClient"];
+        logger.debug(`config/webClient: ${config.modules.Dependent}`);
         if (to.meta.validStates.includes(currentUserState)) {
-            next();
+            if (
```

Comment:
This is too specific, the method checking for required modules needs to be agnostic of what modules are available

Reply:
not sure what you meant, do you have any suggestions?

Reply:
This method is executed for all routes. So having the Dependant module being explicit there is unecessary.
something like this would work:
```
isValidState = to.meta.validStates.includes(currentUserState);
hasRequiredModules = to.meta.requiredModules.every(val => availiableModules.includes(val)))
if(isValidState && hasRequiredModules )
{
   ...
}
``